Dockerfile

```diff
@@ -1,73 +1,131 @@
+COPY --chown=goblin:goblin packages/backend/native-utils/migration/Cargo.toml packages/backend/native-utils/migration/Cargo.toml
+COPY --chown=goblin:goblin packages/backend/native-utils/migration/src/lib.rs packages/backend/native-utils/migration/src/
 
 # Install cargo dependencies
-RUN cargo fetch --locked --manifest-path /firefish/packages/backend/native-utils/Cargo.toml
+RUN cargo fetch --locked --manifest-path /goblin/packages/backend/native-utils/Cargo.toml
+
 
 # Copy only the dependency-related files first, to cache efficiently
-COPY package.json pnpm*.yaml ./
-COPY packages/backend/package.json packages/backend/package.json
-COPY packages/client/package.json packages/client/package.json
-COPY packages/sw/package.json packages/sw/package.json
-COPY packages/firefish-js/package.json packages/firefish-js/package.json
-COPY packages/megalodon/package.json packages/megalodon/package.json
-COPY packages/backend/native-utils/package.json packages/backend/native-utils/package.json
-COPY packages/backend/native-utils/npm/linux-x64-musl/package.json packages/backend/native-utils/npm/linux-x64-musl/package.json
-COPY packages/backend/native-utils/npm/linux-arm64-musl/package.json packages/backend/native-utils/npm/linux-arm64-musl/package.json
+COPY --chown=goblin:goblin package.json pnpm*.yaml ./
+COPY --chown=goblin:goblin packages/backend/package.json packages/backend/package.json
+COPY --chown=goblin:goblin packages/client/package.json packages/client/package.json
+COPY --chown=goblin:goblin packages/sw/package.json packages/sw/package.json
+COPY --chown=goblin:goblin packages/goblin-js/package.json packages/goblin-js/package.json
+COPY --chown=goblin:goblin packages/megalodon/package.json packages/megalodon/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/package.json packages/backend/native-utils/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/npm/linux-x64-musl/package.json packages/backend/native-utils/npm/linux-x64-musl/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/npm/linux-arm64-musl/package.json packages/backend/native-utils/npm/linux-arm64-musl/package.json
 
 # Configure corepack and pnpm, and install dev mode dependencies for compilation
-RUN corepack enable && corepack prepare pnpm@latest --activate && pnpm i --frozen-lockfile
+RUN corepack install -g pnpm@8.12.0
+# && corepack use pnpm@8.12.0
+RUN pnpm i --frozen-lockfile
+# --frozen-lockfile
 
 # Copy in the rest of the native-utils rust files
-COPY packages/backend/native-utils packages/backend/native-utils/
+COPY --chown=goblin:goblin packages/backend/native-utils packages/backend/native-utils/
 
 # Compile native-utils
 RUN pnpm run --filter native-utils build
 
 # Copy in the rest of the files to compile
-COPY . ./
+COPY --chown=goblin:goblin . ./
 RUN env NODE_ENV=production sh -c "pnpm run --filter '!native-utils' build && pnpm run gulp"
```

Comment:
also, offtopic but like. gulp. so many war flashbacks. super low priority obviously but I'd be glad if this old man could get sent ~behind barn~ to greener pastures